### PR TITLE
Fix queue worker incorrectly stopped when there are still more items in the queue

### PR DESCRIPTION
**Pull request by wxiaoguang**
Without `case <-t.C`, the workers would stop incorrectly, the test won't pass. For the worst case, there might be only one running worker processing the queue items for long time because other workers are stopped. Thanks to oliverpool for looking into the problem and proposing a fix.

Actually, the real problem is not completely fixed, the root cause is not directly related to the ticker, but is related to the logic of doDispatchBatchToWorker. It isn't a serious problem at the moment, so keep it as-is. More comments are added.

### CODE REVIEW

File: modules/queue/workergroup.go [hunk 1]
```diff
@@ -143,7 +146,11 @@ func (q *WorkerPoolQueue[T]) doStartNewWorker(wp *workerGroup[T]) {
 		log.Debug("Queue %q starts new worker", q.GetName())
 		defer log.Debug("Queue %q stops idle worker", q.GetName())
 
+		atomic.AddInt32(&q.workerStartedCounter, 1)
```
Comment (lunny) [inline modules/queue/workergroup.go]: When this variable will not be decreased?

Comment (wxiaoguang) [reply modules/queue/workergroup.go]: It never decreases. It's just for debug purpose.

There are other variables like `workerMaxNum` and `workerActiveNum`, they increases/decreases.

**Review by lunny — APPROVED** [review]

File: modules/queue/workergroup.go (hunk 1 above)
Comment (delvh) [reply modules/queue/workergroup.go]: We should probably document this
```suggestion
		atomic.AddInt32(&q.workerStartedCounter, 1) // Only increase counter, used for debugging
```

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
@oliverpool  I see you said you feel "sad" about there is no "@" ping here.

But actually, when I worked with some people from your fork, they said "don't bother me". So, that's really difficult to work with the people from the fork:

* Sometimes: "Oh, don't ping me"
* Sometimes: "Oh, they don't ping me"


And as I said in the description, thank you for looking into the problem, but your fix is not a complete one (neither this PR is). The root problem is the data-race between doDispatchBatchToWorker and workerNum. Maybe you could try to think about a complete solution (no need to thank to me for sharing the idea, feel free to do what you like)

**Pull request merged**